### PR TITLE
Add projects and project index/search

### PR DESCRIPTION
**Pull request by jasonlally**
This is a work in progress, do not merge

@jszwedko - I'll update this later tonight with additional comments as necessary.

Leaving this issue here for later reference: https://github.com/olivernn/lunr.js/issues/38 as I'm having the same problem.

Also this repo: https://github.com/slashdotdash/jekyll-lunr-js-search for inspiration on a pattern

### CODE REVIEW

**Comment by jszwedko** [comment]
@jasonlally Just lemme know when this is ready to look at.

File: _layouts/project.html [hunk 1]
```diff
@@ -23,36 +23,94 @@ <h2>{{ page.title }}</h2>
...
             <td>Website</td>
-            <td><a target='_blank' href='{{page.website}}'>{{page.title}}</a></td>
+            <td><a target='_blank' href='{{page.app_link}}'>{{page.app_link}}</a></td>
           </tr>
           {% endif %}
+          {% if page.repos %}
           <tr>
-            <td>Tags</td>
+            <td>Code</td>
+            <td>{% for repo in page.repos %}{% if forloop.first != true %} <br />{% endif %}<a target='_blank' href='{{repo}}'>{{repo}}</a>{% endfor %}</td>
```
Comment (jszwedko) [inline _layouts/project.html]: Nitpicky, but could we use a `<ul class="unstyled">` rather than `<br />`s?

Comment (jszwedko) [reply _layouts/project.html]: This'll also let you drop the `forloop.first` logic.

File: Gemfile.lock [hunk 2]
```diff
@@ -189,6 +189,8 @@ GEM
       jekyll (>= 3.0)
     kramdown (1.16.2)
     liquid (4.0.0)
+    liquid-md5 (0.0.3)
```
Comment (jszwedko) [inline Gemfile.lock]: Is this being used anywhere? I couldn't find it.

**Review by jszwedko — COMMENTED** [review]
Overall this is looking great! The Github diff isn't view-able since the changeset is too large, but I think I peeked at all the important changes locally.

I assume you are still working on `searchProjects.js`.

File: Gemfile.lock (hunk 2 above)
Comment (jasonlally) [reply Gemfile.lock]: No, it was going to be, but it won't work on github pages.

File: _layouts/project.html (hunk 1 above)
Comment (jasonlally) [reply _layouts/project.html]: Yeah, I'll change this, but I actually want to change the whole table altogether.

**Comment by jasonlally** [comment]
@jszwedko - this is ready for now

It:
1. Creates a searchable list of active projects
2. Creates a layout for project pages
3. Roughly hews to what the user research group has been mocking up w/r/t project matching

I'll want to do a couple of things later:

1. Add pagination to longer lists of project results (slimmed down the list a bit, so not as much a problem right now)
2. Fine tune the search UX

More importantly, we should just have a plan for getting updated content from projects and we should validate we're accommodating the important information on the project pages.

**Comment by jszwedko** [comment]
I'll never get used to those optional semicolons.

Thanks! I agree with your next steps, but I think this is good to go for the beta launch.

We can advertise the project bits over the next few weeks to encourage people to update them via the Markdown and transition into Netlify or similar. This'll dove tail into the project matching work too.

**Pull request merged**